vsm

```diff
@@ -147,31 +149,55 @@ def process(state, signal, value):
     state.got_signal(signal, value)
 
 
-def run(state):
+def receive():
     '''
-        Loop to grab logic from stdin, then spawn a threads for processing
+        Loop to grab logic from stdin
     '''
     for line in sys.stdin:
         line = line.replace(" ", "").strip('\n')
         if line == "":
             # Ignore blank lines
             pass
         if line == "quit":
-            break
+            exit(0)
         elif "=" in line:
             line = line.split("=")
-            process(state, line[0], line[1])
+            return (line[0], line[1])
         else:
-            process(state, line, None)
-
+            return (line, None)
+
+def run(state, no_loop):
+    try:
+        while True:
+            message = receive()
+            if message == None:
+                exit(0)
+            signal, value = message
+            process(state, signal, value)
+            if no_loop:
+                break
+    except KeyboardInterrupt:
+        exit(0)
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument('--initial-state', type=str,
                         help='Initial state, yaml file', required=False)
     parser.add_argument('rules', type=str,
                         help='yaml rules configuration')
+    parser.add_argument('--ipc-module', type=str, help='Load IPC module')
```

Comment:
We should list the modules that are available and/or provide an option that does that. That could be a new task, though =)